include/cuco/extent.cuh

```diff
@@ -120,30 +105,9 @@ struct extent<SizeType, dynamic_extent> {
  * @return Resulting valid extent
  */
 template <int32_t CGSize, int32_t WindowSize, typename SizeType, std::size_t N>
-[[nodiscard]] auto constexpr make_valid_extent(extent<SizeType, N> ext)
-{
-  auto constexpr max_prime = cuco::detail::primes.back();
-  auto constexpr max_value =
-    (static_cast<uint64_t>(std::numeric_limits<SizeType>::max()) < max_prime)
-      ? std::numeric_limits<SizeType>::max()
-      : static_cast<SizeType>(max_prime);
-  auto const size = SDIV(ext, CGSize * WindowSize);
-  if (size <= 0 or size > max_value) { CUCO_FAIL("Invalid input extent"); }
-
-  if constexpr (N == dynamic_extent) {
-    return extent<SizeType>{static_cast<SizeType>(
-      *cuco::detail::lower_bound(
-        cuco::detail::primes.begin(), cuco::detail::primes.end(), static_cast<uint64_t>(size)) *
-      CGSize)};
-  }
-  if constexpr (N != dynamic_extent) {
-    return extent<SizeType,
-                  static_cast<std::size_t>(*cuco::detail::lower_bound(cuco::detail::primes.begin(),
-                                                                      cuco::detail::primes.end(),
-                                                                      static_cast<uint64_t>(size)) *
-                                           CGSize)>{};
-  }
-}
+[[nodiscard]] auto constexpr make_valid_extent(extent<SizeType, N> ext);
```

Comment:
Just to double check, is this function invokable on the device side?

Reply:
That's an excellent topic! The problem I'm currenrly facing wirh making this function callable from the device, too, is that it needs access to the prime array which is deckared `inline constexpr`, which means it's only accessible from the host. We could declared it e.g. `__constant__` but it would be a runtime constant.